walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,118 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
 
 public final class FindMeetingQuery {
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    
+    /* Initializing an array that will provide information for each minute in a day. 
+       dayMinutes[j] will represent what is happening in minute j, where j is
+       from 0 to 3599. */
+    ArrayList<MinuteDetails> dayMinutes = new ArrayList<>();
+    for (int i = 0; i <= TimeRange.END_OF_DAY  + 1; i++) {
```

Comment:
Apparently END_OF_DAY is considered 23:59, and the way they (and myself also) constructed the intervals was that they considered the ending of an interval the minute after it actually ends, for example if you have a meeting starting at 23:00 at it lasts for an hour, the ending is set at 00:00, which is the 1440th minute in the day.

Reply:
Right, but unless you need to schedule a meeting of duration zero, the last valid `start_time` is still END_OF_DAY. You don't really need to record any info at time END_OF_DAY or END_OF_DAY+1 since there is no valid meeting which could possible start then.

Reply:
Oh, ya, that makes sense, I will change